src/DependencyInjection/Handler/HandlerExtensionInterface.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MonologBundle\DependencyInjection\Handler;
+
+use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
+use Symfony\Component\DependencyInjection\Definition;
+
+interface HandlerExtensionInterface
+{
+    /**
+     * Returns the name of the handler and configuration subtree.
+     */
+    public function getName(): string;
+
+    /**
+     * Add properties for the handler type.
+     */
+    public function buildArrayNode(ArrayNodeDefinition $handlerNode): void;
+
+    /**
+     * @param array{
+     *     level: string|int,
+     *     bubble: bool,
+     *     formatter: string|null,
+     * } $config Generic options
+     * @param array{} $handler Specific handler options
```

Comment:
this is wrong. `array{}` is the shape representing an empty array.

Reply:
This is on purpose. By default this is an empty array. You have to specify another array shape in subclasses.

Reply:
That's not the proper way to describe such API. We would need to use a generic type with `THandlerConfig of array<string, mixed>`, with child classes describing their array shape as the template type.